File: frontend/apps/web/src/types/moments.ts
/**
 * Moments (朋友圈) TypeScript type definitions
 * Matches Rust backend DTOs with serde(rename_all = "camelCase")
 */

export interface MomentPost {
  id: string;
  userId: string;
  content?: string;
  visibility: 0 | 1 | 2; // 0=public, 1=friends, 2=private
  linkUrl?: string;
  linkTitle?: string;
  linkCover?: string;
  location?: string;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
}

export interface MomentMedia {
  id: string;
  postId: string;
  type: 0 | 1; // 0=image, 1=video
  url: string;
  sortOrder: number;
}

export interface MomentLike {
  id: string;
  postId: string;
  userId: string;
  createdAt: string;
  nickname?: string;
  avatar?: string;
}

export interface MomentComment {
  id: string;
  postId: string;
  userId: string;
  parentId?: string;
  content: string;
  createdAt: string;
  nickname?: string;
  avatar?: string;
}

export interface MomentNotification {
  id: string;
  userId: string;
  actorId: string;
  notificationType: 'like' | 'comment';
  postId: string;
  commentId?: string;
  isRead: boolean;
  createdAt: string;
  actorNickname?: string;
  actorAvatar?: string;
}

export interface PostWithDetails {
  post: MomentPost;
  media: MomentMedia[];
  likeCount: number;
  commentCount: number;
  isLiked: boolean;
  userNickname?: string;
  userAvatar?: string;
}

export interface CreatePostRequest {
  content?: string;
  visibility?: 0 | 1 | 2;
  linkUrl?: string;
  linkTitle?: string;
  linkCover?: string;
  location?: string;
}

export interface CreateCommentRequest {
  content: string;
  parentId?: string;
}

export interface FeedQuery {
  cursor?: string;
  limit?: number;
}

File: rust/crates/im-common/src/moments.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MomentPost {
    pub id: i64,
    pub user_id: i64,
    pub content: Option<String>,
    pub visibility: i8,
    pub link_url: Option<String>,
    pub link_title: Option<String>,
    pub link_cover: Option<String>,
    pub location: Option<String>,
    pub status: i8,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MomentMedia {
    pub id: i64,
    pub post_id: i64,
    pub media_type: i8,
    pub url: String,
    pub sort_order: i8,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MomentLike {
    pub id: i64,
    pub post_id: i64,
    pub user_id: i64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MomentComment {
    pub id: i64,
    pub post_id: i64,
    pub user_id: i64,
    pub parent_id: Option<i64>,
    pub content: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MomentNotification {
    pub id: i64,
    pub user_id: i64,
    pub actor_id: i64,
    pub notification_type: String,
    pub post_id: i64,
    pub comment_id: Option<i64>,
    pub is_read: bool,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreatePostRequest {
    pub content: Option<String>,
    pub visibility: i8,
    pub link_url: Option<String>,
    pub link_title: Option<String>,
    pub link_cover: Option<String>,
    pub location: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateCommentRequest {
    pub content: String,
    pub parent_id: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PostWithDetails {
    pub post: MomentPost,
    pub media: Vec<MomentMedia>,
    pub like_count: i64,
    pub comment_count: i64,
    pub is_liked: bool,
    pub user_nickname: Option<String>,
    pub user_avatar: Option<String>,
}
